mzLib/MassSpectrometry/IsobaricQuantSampleInfo.cs

```diff
@@ -0,0 +1,251 @@
+        /// </summary>
+        /// <param name="obj">The object to compare.</param>
+        /// <returns>True if the object is an <see cref="IsobaricQuantSampleInfo"/> or <see cref="ISampleInfo"/>
+        /// with matching <see cref="FullFilePathWithExtension"/> and <see cref="ChannelLabel"/>; otherwise false.</returns>
+        public override bool Equals(object? obj)
+        {
+            if (obj is null) return false;
+            if (obj is IsobaricQuantSampleInfo otherIsobaric)
+            {
+                return Equals(otherIsobaric);
+            }
+            if (obj is ISampleInfo otherSampleInfo)
+            {
+                return Equals(otherSampleInfo);
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Returns a hash code based on <see cref="FullFilePathWithExtension"/> and <see cref="ChannelLabel"/>,
+        /// consistent with the equality comparison used in <see cref="Equals(IsobaricQuantSampleInfo?)"/>.
+        /// </summary>
+        /// <returns>A hash code for this instance.</returns>
+        public override int GetHashCode()
+        {
+            return HashCode.Combine(
+                StringComparer.Ordinal.GetHashCode(FullFilePathWithExtension ?? string.Empty),
+                StringComparer.Ordinal.GetHashCode(ChannelLabel));
+        }
+
+        /// <summary>
+        /// Returns a string representation of this sample in the format "{FileName}_{ChannelLabel}".
+        /// Uses only the file name (without directory path) for readability.
+        /// </summary>
+        /// <returns>A display string identifying this sample by file name and channel.</returns>
+        public override string ToString()
+        {
+            var fileName = Path.GetFileNameWithoutExtension(FullFilePathWithExtension);
+            return $"{fileName}_{ChannelLabel}";
+        }
+
+        /// <summary>
+        /// Compares this instance to another <see cref="ISampleInfo"/> for sorting purposes.
+        /// For <see cref="IsobaricQuantSampleInfo"/> instances, compares by <see cref="FullFilePathWithExtension"/> first,
+        /// then by <see cref="ChannelLabel"/>. All comparisons use ascending order 
+        /// (A before B for strings). Non-null values sort before null.
+        /// 
+        /// When comparing to a non-isobaric <see cref="ISampleInfo"/>, isobaric samples sort after
+        /// other sample types.
+        /// </summary>
+        /// <param name="other">The other <see cref="ISampleInfo"/> to compare to.</param>
+        /// <returns>
+        /// A negative value if this instance precedes <paramref name="other"/>;
+        /// zero if they are equal; a positive value if this instance follows <paramref name="other"/>.
+        /// </returns>
+        public int CompareTo(ISampleInfo? other)
+        {
+            // Non-null comes before null
+            if (other is null) return -1;
+
+            if (other is IsobaricQuantSampleInfo otherIsobaric)
```

Comment:
Would it make more sense to compare based on Condition, then Biorep, then Fraction, then TechRep?